File: Day1/secrete_entrance.py
import sys

def main():
    pos = 50    # posición inicial del dial
    count = 0   # cuántas veces termina en 0

    tokens = sys.stdin.read().split()

    for token in tokens:
        token = token.strip()
        if not token:
            continue  # saltar líneas vacías

        direction = token[0]      # 'L' o 'R'
        steps_str = token[1:]     # el número como texto

        try:
            steps = int(steps_str)
        except ValueError:
            print(f"Error: no pude convertir {steps_str!r} a entero (línea: {token!r})")
            return

        if direction == 'R':
            pos += steps
        elif direction == 'L':
            pos -= steps
        else:
            print(f"Error: dirección inválida {direction!r} en línea {token!r}")
            return

        # Normalizar a rango 0..99 (maneja negativos)
        pos = (pos % 100 + 100) % 100

        if pos == 0:
            count += 1

    print(count)


if __name__ == "__main__":
    main()

File: Day1/secrete_entrance.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	pos := 50   // posición inicial del dial
	count := 0 // cuántas veces pasa por 0

	data, err := os.ReadFile("input.txt")
	if err != nil {
		fmt.Println("Error al leer input.txt:", err)
		return
	}

	tokens := strings.Fields(string(data))

	for _, token := range tokens {
		direction := token[0]   // 'L' o 'R'
		stepsStr := token[1:]   // número

		var steps int
		_, err := fmt.Sscanf(stepsStr, "%d", &steps)
		if err != nil {
			fmt.Println("Error al convertir:", stepsStr)
			return
		}

		delta := 0
		if direction == 'R' {
			delta = 1
		} else if direction == 'L' {
			delta = -1
		} else {
			fmt.Println("Dirección inválida:", token)
			return
		}

		// Avanzar paso por paso
		for i := 0; i < steps; i++ {
			pos += delta
			pos = (pos + 100) % 100
			if pos == 0 {
				count++
			}
		}
	}

	fmt.Println(count)
}
